scalanet/src/io/iohk/scalanet/peergroup/UDPPeerGroup.scala

```diff
@@ -98,31 +95,25 @@ class UDPPeerGroup[M](val config: Config, cleanupScheduler: Scheduler = Schedule
           .addLast(new channel.ChannelInboundHandlerAdapter() {
             override def channelRead(ctx: ChannelHandlerContext, msg: Any): Unit = {
               val datagram = msg.asInstanceOf[DatagramPacket]
+              val remoteAddress = datagram.sender()
+              val localAddress = datagram.recipient()
+              log.info(s"Client channel read message with remote $remoteAddress and local $localAddress")
+              val udpChannelId = UdpChannelId(ctx.channel().id(), remoteAddress, localAddress)
               try {
-                val remoteAddress = datagram.sender()
-                val localAddress = datagram.recipient()
-                log.info(s"Client channel read message with remote $remoteAddress and local $localAddress")
-
-                val channelId = (remoteAddress, localAddress)
-
-                val channel = activeChannels.get(channelId)
-
-                if (channel == null || !channel.isOpen) {
-                  // It should never happen as closing client channel closes underlying netty channel, so channelRead should
-                  // not execute on this channel
-                  throw new IllegalStateException(s"Missing channel instance for channelId $channelId")
-                } else {
-                  handleIncomingMessage(channel, datagram)
-                }
+                println(s"Received message from channel ${udpChannelId}")
+                Option(activeChannels.get(udpChannelId)).foreach(handleIncomingMessage(_, datagram))
               } catch {
-                case NonFatal(e) => handleError(ctx, e)
+                case NonFatal(e) => handleError(udpChannelId, e)
```

Comment:
would it throw now?

Reply:
i think there is still possiblity to throw on `datagramPacket.content().nioBuffer()` if for some reason  someone would start udppeergroup with bytebuffer which does not have such ability. 
From dosc about this function:
```
* @throws UnsupportedOperationException
     *         if this buffer cannot create a {@link ByteBuffer} that shares the content with itself
```